src/pages/Analytics/analyticsDatastoreQuery.js

```diff
@@ -32,15 +33,38 @@ export const getAnalyticsKeyQuery = {
  * */
 
 export const useReadAnalyticsDataStore = () => {
-    const { loading, error, data } = useDataQuery(getAnalyticsKeyQuery)
+    const [teiAnalytics, setTei] = useState()
+    const [visualizations, setVisualizations] = useState()
+    const [home, setHome] = useState()
+    const [program, setProgram] = useState()
+    const [dataSet, setDataSet] = useState()
+
+    const { loading, error } = useDataQuery(getAnalyticsKeyQuery, {
+        onComplete: result => {
```

Comment:
Because of [the way app-runtime v3 caches queries](https://developers.dhis2.org/blog/2021/11/app-platform-v8#new-feature-usedataquery-caches-and-deduplicates-queries), using onComplete in this way would make it hard to migrate this app to app platform v8/app runtime v3 in the future.

Two ways to refactor this would be:
- Revert to previous method of returning relevant fields from `data`
- Move code inside `onComplete` to a `useEffect` hook with `data` as its dependency

Reply:
I'm refactoring it like this: 
```
useEffect(() => {
        if (data) {
            const { tei, dhisVisualizations } = data.analytics
            setTei(tei || [])
            setHome(dhisVisualizations ? dhisVisualizations.home : [])
            setProgram(dhisVisualizations ? dhisVisualizations.program : {})
            setDataSet(dhisVisualizations ? dhisVisualizations.dataSet : {})
            setVisualizations(
                dhisVisualizations
                    ? dhisVisualizations
                    : {
                        home: [],
                        program: {},
                        dataSet: {},
                    }
            )
        }
    }, [data])
```